### ISSUE TITLE
HTX: all private watch functions don't work because of encoding issues

### ISSUE THREAD

**Issue by mjwvb** [ccxt/ccxt#22451]
### Operating System

Ubuntu

### Programming Languages

JavaScript

### CCXT Version

4.3.18

### Description

When using watchOrders(), watchMyTrades(), watchBalance(), and probably others, I get a lot of unreadable gzip encoding errors. Stuff like ��%���O;<D/=�=��T. Checked the docs of HTX, and it says "All return data of websocket Market APIs are compressed with GZIP so they need to be unzipped.". See: https://www.htx.com/en-us/opend/newApiPages/?id=662

I uncommented the following line in the ccxt and now it seems to work perfectly fine:
https://github.com/ccxt/ccxt/blob/e4581cd6a34f9db7d43da9f89a2828c8d2fc42bd/js/src/pro/htx.js#L2312

I guess this gzip exception in the code is no longer needed and should be enabled for all websocket calls.

**Comment by sc0Vu** [comment]
@mjwvb thanks, we'll update this asap. BTW, could you overwrite the ws options before we fix?

**Comment by carlosmiei** [comment]
Hello @mjwvb, what's the symbol used? 

Locally spot seems ok
```
 n huobi watchOrders          
2024-05-10T10:35:18.231Z
Node.js: v18.18.0
CCXT v4.3.18
huobi.watchOrders ()
              id |                                  clientOrderId | lastTradeTimestamp |   status |   symbol |  type | side | price | amount | filled | remaining | cost | trades | fees
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
1060375938336019 | AA03022abcf2c27a72-76a9-404a-8db8-adc58ad30b67 |      1715337347157 | canceled | LTC/USDT | limit |  buy |    50 |    0.2 |      0 |       0.2 |    0 |     [] |   []
1 objects
```

**Comment by carlosmiei** [comment]
hum, swap too
```
n huobi watchOrders "LTC/USDT:USDT" 
2024-05-10T10:37:19.097Z
Node.js: v18.18.0
CCXT v4.3.18
huobi.watchOrders (LTC/USDT:USDT)
                 id | lastTradeTimestamp | status |        symbol |  type | side | price | amount |                            fee | trades |                           fees
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
1238560575692472320 |      1715337452607 |   open | LTC/USDT:USDT | limit |  buy |    50 |      1 | {"cost":"0","currency":"USDT"} |     [] | [{"cost":0,"currency":"USDT"}]
1 objects
```

**Comment by mjwvb** [comment]
On various spot pairs, e.g. BTC/USDT.

Getting a lot of these in my console:

TypeError: Cannot use 'in' operator to search for 'id' in �V�LQ�R2V�Q*.I,)-r�A�Ҥ�T�TnbQvj�^RIriqJ�^nR����PE	P�������������e-W�M
    at htx.handleMessage (file:///.../node_modules/ccxt/js/src/pro/htx.js:2021:22)
    at WsClient.onMessage (file:///.../node_modules/ccxt/js/src/base/ws/Client.js:297:18)
    at callListener (/.../node_modules/ccxt/node_modules/ws/lib/event-target.js:290:14)
    at WebSocket.onMessage (/.../node_modules/ccxt/node_modules/ws/lib/event-target.js:209:9)
    at WebSocket.emit (node:events:527:28)
    at WebSocket.emit (node:domain:475:12)
    at Receiver.receiverOnMessage (/.../node_modules/ccxt/node_modules/ws/lib/websocket.js:1209:20)
    at Receiver.emit (node:events:527:28)
    at Receiver.emit (node:domain:475:12)
    at /.../node_modules/ccxt/node_modules/ws/lib/receiver.js:581:16
    at node:internal/process/task_queues:141:7
    at AsyncResource.runInAsyncScope (node:async_hooks:202:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:138:8)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    
This is on macos with nodejs v20.12.2. This also happens on my ubuntu trading bot server.

**Comment by mjwvb** [comment]
Also just tried on nodejs v18.19.0, same issue

**Comment by carlosmiei** [comment]
@mjwvb Are you providing a different hostname or special configs to the exchange?

**Comment by mjwvb** [comment]
Excuse me, my errors now occur only when using watchTrades() and watchOrderBook(), which are the public endpoints. Very strange why I posted otherwise yesterday...

**Comment by mjwvb** [comment]
> @mjwvb Are you providing a different hostname or special configs to the exchange?

No special configs on my local test machine. My server machine has the special AWS hostname. The symbols I'm watching are BTC/USDT, ETH/BTC, ETH/USDT, LTC/BTC, LTC/USDT. All five have the same problem

**Comment by mjwvb** [comment]
I'm trying to isolate the issues and to reproduce it in a simple code snippet. Somehow then I can't reproduce it, so there's maybe a problem on my side, or a combination of setups that's causing the issue. I'll post here when I have isolated the issue.

**Comment by mjwvb** [comment]
@carlosmiei Got it. The problem occurs when you combine private and public websocket apis. Sorry for not providing an isolated testable code snippet initially, but here it is:

```typescript
const test = async () => {
  const exchange = new ccxt.pro.htx({
    apiKey: process.env.EXCHANGE_HTX_API_KEY,
    secret: process.env.EXCHANGE_HTX_API_SECRET,
  });

  const watchTrades = async (symbol: string) => {
    while (true) {
      try {
        console.log("Watching trades:", symbol);
        await exchange.watchTrades(symbol);
      } catch (e) {
        console.log(e);
        if (e instanceof ccxt.BadRequest) throw e;
      }
    }
  };

  const watchBalance = async () => {
    while (true) {
      try {
        console.log("Watching balance");
        await exchange.watchBalance();
      } catch (e) {
        console.log(e);
        if (e instanceof ccxt.BadRequest) throw e;
      }
    }
  };

  watchTrades("BTC/USDT");
  watchBalance();
};

test();
```

**Comment by carlosmiei** [comment]
@mjwvb Oh, that's even weirder 😅; we will take a look, but in the meantime, as a workaround, you can create two instances like `htxPrivate` and `htxPublic`, and that should be enough to unblock you for now, right?

**Comment by mjwvb** [comment]
Yeah that works for now, thanks!

**Comment by pcriadoperez** [comment]
Hi @mjwvb, in the latest ccxt version, using the script it seems to be working normally for me. Are you able to reproduce consistently with the script?

**Comment by mjwvb** [comment]
Hello @pcriadoperez, you're right... Now I'm very confused... Anyway, when I run watchBalance() before watchTrades("BTC/USDT"), then the encoding issues are consistently back again. So it seems the private api must be initiated before the public api.

Maybe there is/was some racing going on, because I'm pretty certain the script worked this way before.

```typescript
watchBalance();
watchTrades("BTC/USDT");
```
